### PR TITLE
ISPN-8843 Workaround for the IDE compiler

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-8843

### CODE REVIEW

**Comment by gustavonalle** [comment]
@karesti Kind of what you suggested earlier, but making it optional and adding a build extension to obtain the os

**Comment by karesti** [comment]
@gustavonalle it works, for me should be merged til we find another solution
@slaskawi any additional thoughts ?

**Comment by gustavonalle** [comment]
CI is fine

File: server/core/pom.xml [hunk 1]
```diff
@@ -23,6 +23,22 @@
          <groupId>io.netty</groupId>
          <artifactId>netty-handler</artifactId>
       </dependency>
+      <!-- Workaround for ISPN-8843 -->
+      <dependency>
+         <groupId>org.conscrypt</groupId>
+         <artifactId>conscrypt-openjdk</artifactId>
+         <version>${version.netty-conscrypt-optional}</version>
+         <classifier>${os.detected.classifier}</classifier>
```
Comment (rvansa) [inline server/core/pom.xml]: Shouldn't this be in test scope?

Comment (gustavonalle) [reply server/core/pom.xml]: let me try it and see it Intellij is happy

Comment (slaskawi) [reply server/core/pom.xml]: It doesn't really matter since this is an `<optional>true</optional>` dependency.

File: server/core/pom.xml [hunk 2]
```diff
@@ -59,6 +75,14 @@
    </dependencies>
 
    <build>
+      <!-- Workaround for ISPN-8843 -->
+      <extensions>
+         <extension>
+            <groupId>kr.motd.maven</groupId>
+            <artifactId>os-maven-plugin</artifactId>
+            <version>1.4.1.Final</version>
```
Comment (slaskawi) [inline server/core/pom.xml]: Version should probably be in the BOM or parent pom.

File: server/core/pom.xml (hunk 1 above)
Comment (gustavonalle) [reply server/core/pom.xml]: it doesn't work

**Comment by gustavonalle** [comment]
> Let's make a deal than - let's wait a couple of days (a week?)

IMO we should integrate this now and revert when (if) they fix it.

File: server/core/pom.xml (hunk 2 above)
Comment (gustavonalle) [reply server/core/pom.xml]: Let's not make this workaround official :)

**Comment by gustavonalle** [comment]
>  But I do understand that you guys want to import everything into IntelliJ and just run some tests...

I didn't know you could import only certain modules in Intellij. I usually just go to File -> Open and it materializes the maven/sbt/gradle model

**Comment by slaskawi** [comment]
@gustavonalle Yes you can! Just open whatever project you need (or start with an empty one), bring up Maven window and point `client/hotrod-client/pom.xml`. This way you will import only one module. I've always been working that way - IntelliJ indexing is killing me so I open up only those modules that I'm working on.

**Comment by karesti** [comment]
@slaskawi @gustavonalle I'm with Gustavo. We merge and revert after. It's not a big deal this modification in the pom and the problem is super annoying

File: server/core/pom.xml (hunk 2 above)
Comment (gustavonalle) [reply server/core/pom.xml]: done

**Comment by galderz** [comment]
Merged, this is a roadblock for all of us. Create follow up JIRA(s) for any leftover work.

**Comment by karesti** [comment]
@galderz thank you !!!

**Pull request merged**